examples/chart/teleport-agent/templates/config.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}
+data:
+  teleport.yaml: |
+    teleport:
+      auth_token: {{ required "authToken is required in chart values" .Values.authToken }}
```

Comment:
I would consider creating a Secret with the token instead, mounting it into the container and taking it from there, I think Teleport supports that: https://goteleport.com/teleport/docs/admin-guide/#static-tokens.

Reply:
This complicates the setup (one extra object, layer of indirection), so I'd like to skip that.
AFAIK, secrets don't provide any real security benefit, except for hinting to humans that it's a sensitive resource.
My knowledge may be out-of-date, so please correct me if it's wrong :)

Reply:
I think this is fine for now and in a future version we can head down the route of dynamic tokens and some other form of provisioning. People who desperately want additional security at the cost of added complexity now can always fork the chart and come up with their own distribution method.

Reply:
> AFAIK, secrets don't provide any real security benefit, except for hinting to humans that it's a sensitive resource.

This is true for default configurations but a couple of things to consider are:

* Cluster may be configured with a KMS provider in which case secrets data will be encrypted at rest.
* Placing sensitive data in secrets lets users to configure granular RBAC policies for example allowing somebody to view configuration but not the secrets.

It may not be the end of the world as it is (esp. if we're gonna change this eventually to use PKI), I guess I personally just don't see a compelling argument _against_ placing it into a secret. An additional object shouldn't be any trouble as it's all managed by Helm.

Feel free to leave as is if you still feel it's not worth it.

Reply:
RBAC and encryption at rest for Secrets are very good reasons, thanks for pointing those out!
Updated to load join token from a Secret.